### ISSUE TITLE
sap_hana_install role is not picking sap_hana_install_sidadm_password  when master_password is set to n.

### ISSUE THREAD

**Issue by kallass** [sap-linuxlab/community.sap_install#374]
I did not want to set the master password for the installation because I dont want sidam and system password to be same password.
So I set  sap_hana_install_sidadm_password and  sap_hana_install_system_password explicitly in my playbook.

But looks like it does not pass the variables to the hdblcm config file. The install still fails with Mandatory parameter 'password' (Password) is missing or invalid.  
I went through the code and I see no references to sidadm and system password anywhere in the code.

Is it mandatory to set master password or which files send the sid and system password to the config file.

 - hosts: lab1 
  collections:
    - community.sap_install
  become: true
  vars:
    sap_hana_install_system_roles_collection: 'redhat.rhel_system_roles'
    sap_hana_install_software_directory: /hana/software
    sap_hana_install_sapcar_filename: SAPCAR
    sap_hana_install_sarfiles:
      - IMDB_SERVER20_064_0-80002031.SAR
      - IMDB_CLIENT20_013_22-80002082.SAR
    sap_hana_install_modify_selinux_labels: no
    sap_hana_install_use_master_password: 'n'
    sap_hana_install_hdblcm_extraargs: '--ignore=check_signature_file'
    sap_hana_install_sidadm_password: 'Mypass123'
    sap_hana_install_db_system_password: 'password1'
    sap_hana_install_sid: 'SD1'
    sap_hana_install_instance_number: '00'
    sap_hana_install_system_usage: 'test'
    sap_hana_install_check_sidadm_user: no
    sap_hana_install_home: /usr/sap/${sid}
  roles:
    - sap_hana_install

**Comment by kallass** [comment]
In our org, the sidadm user is created by server team so it is skipped in the install . 

One workaround is to put the master password same as sidadm,but then even system password defaults to master password.
So I am trying to avoid the workaround.

**Comment by s11s11** [comment]
@kallass I also ran into this issue since we also have different passwords for sidadm and system.
I noticed that there are two different variables that are related to the system user password:

- sap_hana_install_db_system_password
- sap_hana_install_system_user_password

The first originates from the main.yml file, which has the default parameters for this role. The second originates from the config file template provided by HDBLCM. The role uses the first one as the password when connecting to hdbsql to set the log mode, and the second one as the actual password for the system user.

It's a bit confusing, but that's how I got it to work.

**Comment by kallass** [comment]
@s11s11 From what I realized is that the master password is mandatory. I am yet to figure out how to skip it. so i set the master password same as sidadm password.

i will try this option for system password so that I dont have to change it manually after the install.

**Comment by kallass** [comment]
By setting the following parameters,i could finally skip master password requirement.

    sap_hana_install_use_master_password: 'n'   ## Keep this as n if you dont want to use master password
    sap_hana_install_master_password:      ##Keep this parameter in the playbook but leave it empty
    sap_hana_install_password: MySecret   ##Use this instead of sap_hana_install_sidadm_password. Keep it same as sidadm pwd
    sap_hana_install_db_system_password: MySystem123   ##This is the system password
    sap_hana_install_system_user_password: MySystem123	##keep Same as system password

**Comment by s11s11** [comment]
I'm glad you found a workaround. I still think that the two different parameters for the system user password should be merged, or at least documented proerly.

**Comment by kallass** [comment]
@s11s11 I agree. I tried to put those in my comments so that if someone is searching for the same issue can figure it out easier.

**Comment by berndfinger** [comment]
Let's keep this open until I (or someone else) has fixed it or modified the doc. Thanks @kallass @s11s11 for your investigation!
